### PR TITLE
Some code-cleanup/simplification in apps

### PR DESCRIPTION
**Pull request by richsalz**
Remove some casts on password callback by adding a wrapper function.
Remove level of indent by doing an early-return on failure.

Only worth doing in master IMO.  All tests pass.

### CODE REVIEW

File: apps/apps.c [hunk 1]
```diff
@@ -280,78 +280,78 @@ const UI_METHOD *get_ui_method(void)
...
+
+    if ((ui = UI_new_method(ui_method)) == NULL)
+        return 0;
 
-        ui_flags |= UI_INPUT_FLAG_DEFAULT_PWD;
-        UI_ctrl(ui, UI_CTRL_PRINT_ERRORS, 1, 0, 0);
+    if (cb_data != NULL && cb_data->prompt_info != NULL)
+        prompt_info = cb_data->prompt_info;
+    prompt = UI_construct_prompt(ui, "pass phrase", prompt_info);
+    if (!prompt) {
```
Comment (t-j-h) [inline apps/apps.c]: If you are fixing code then this should be prompt != NULL

File: apps/apps.h [hunk 2]
```diff
@@ -437,6 +437,8 @@ typedef struct pw_cb_data {
     const char *prompt_info;
 } PW_CB_DATA;
 
+/* Sadly, we need both wrap and the "real" function; libcrypto uses both. */
```
Comment (t-j-h) [inline apps/apps.h]: Adding "Sadly" into a comment isn't appropriate.

**Comment by richsalz** [comment]
Added a commit to address Tim's feedback.

File: apps/apps.h (hunk 2 above)
Comment (richsalz) [reply apps/apps.h]: "git grep -i sadly"

**Comment by richsalz** [comment]
ping to merge?

**Comment by levitte** [comment]
Squashed and merged.

229446dfae66aed32a738063c9062fea8e7049dc Some code-cleanup/simplification in apps

**Pull request closed without merging**